src/VirtualFile/VirtualPath.cs

```diff
@@ -29,9 +34,11 @@ public VirtualPath Parent
             return result + "/";
         }
     }
-    public VirtualPath(string path)
+
+    public VirtualPath(string path, IFileProvider fileProvider = null)
```

Comment:
I'd rather the methods that need IFileProvider be updated to take one - it'll make it clear where it's actually needed